File: LeetCode_540/main.cpp
/**
    LeetCode_540
    https://leetcode.com/problems/single-element-in-a-sorted-array/description/
    MatrixDeity, 2018.
*/

#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        int i = 0, j = nums.size() - 1;
        while (i <= j) {
            int m = i + (j - i) / 2;
            if (nums[m] == nums[m ^ 1])
                i = m + 1;
            else
                j = m - 1;
        }
        return nums[i];
    }
};

int main() {
    return 0;
}

File: LeetCode_540/main.go
/**
  LeetCode_540
  https://leetcode.com/problems/single-element-in-a-sorted-array/
  MatrixDeity, 2023.
*/

package leetcode540

func singleNonDuplicate(nums []int) int {
	i, j := 0, len(nums)-1

	for i < j {
		m := i + (j-i)/2
		if nums[m] == nums[m^1] {
			i = m + 1
		} else {
			j = m
		}
	}

	return nums[i]
}
